### PR TITLE
Feature/academy

### PR DESCRIPTION
**Pull request by danholdaway**
## Description

- Let user specify number of bins in the innovation plots
- Update the convergence diagnostic to work with DRPCG minimizer instead of DRIPCG minimizer
- Add diagnostic for plotting pie charts of timing stats needed in Thursday model tutorial.

4dvar_pseuso_lowresinc_method_total_time_20201001_000000
4dvar_pseuso_method_total_time_20201001_000000

### CODE REVIEW

File: src/Workflows/Generic/Diagnostics/log_timing.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+application:
+
+  # Application to use
+  application name: log_timing
+
+  # JEDI log file to read
+  log file: /gpfsm/dnb31/drholdaw/JediWF/fv3-jedi-tools-testing/hyb-3dvar_20180414_000000.run
+
+  # Number of methods to show in the pie chart. Code will pick n most expensive (default is 9)
```
Comment (shlyaeva) [inline src/Workflows/Generic/Diagnostics/log_timing.yaml:9]: default is 10, I think, from looking at the code.

**Review by shlyaeva — APPROVED** [review]
I really like the new plot, neat!
I have not reviewed `diag_log_timing.py` script, only reviewed all other changes.

**Review by mmiesch — APPROVED** [review]
I haven't tested it yet but the plots look great!

**Comment by shlyaeva** [comment]
I'm trying this out with the updated tutorial, and drpcg_norm_gradient looks good, but drpcg_normalized-gradient-reduction plots iteration number.

**Comment by danholdaway** [comment]
drpcg_norm_gradient might be a residual file created previously. I'll take a look at that

**Comment by shlyaeva** [comment]
@danholdaway ignore my previous comment; the wrong plot is from the previous run, when I ran with develop.

**Comment by shlyaeva** [comment]
And I mixed them up in the previous comment too, so the new plot looks good.

**Review by shlyaeva — APPROVED** [review]

**Comment by danholdaway** [comment]
Thanks for checking everything is working correctly.

**Pull request merged**